projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -455,30 +511,35 @@ public function parse_numeric_get_arg( $arg ) {
 	 * @return array
 	 */
 	public function get_options() {
-		if ( null === $this->_options ) {
-			$this->_options = Jetpack_Options::get_option( 'relatedposts', array() );
-			if ( ! is_array( $this->_options ) )
-				$this->_options = array();
-			if ( ! isset( $this->_options['enabled'] ) )
-				$this->_options['enabled'] = true;
-			if ( ! isset( $this->_options['show_headline'] ) )
-				$this->_options['show_headline'] = true;
-			if ( ! isset( $this->_options['show_thumbnails'] ) )
-				$this->_options['show_thumbnails'] = false;
-			if ( ! isset( $this->_options['show_date'] ) ) {
-				$this->_options['show_date'] = true;
+		if ( null === $this->options ) {
```

Comment:
I wonder if this could be a good opportunity to move to `wp_parse_args` here, to clean things up a bit?

Reply:
Took a stab at this in f886dc9db1a07ec8df1e1a98f7207a7dc9235883

Reply:
Actually @jeherve looks like that breaks things, but I'm not too sure why. Spent a good bit of time with it, but not sure what I'm doing wrong. The arrays don't seem to be merging. `$this->options` is the same both before and after the merge and the `size` key isn't being set to `3`.